src/libcec/CECClient.cpp

```diff
@@ -214,7 +214,12 @@ bool CCECClient::SetHDMIPort(const cec_logical_address iBaseDevice, const uint8_
     }
   }
 
-  return SetPhysicalAddress(iPhysicalAddress);
+  // and set the address
+  SetDevicePhysicalAddress(iPhysicalAddress);
+
+  QueueConfigurationChanged(m_configuration);
```

Comment:
this doesn't look right, or at least not the correct place to do this.
I'll have to look into this in more detail

Reply:
@opdenkamp I do agree there might be a better way of doing this. At the same time, that's how it was implemented 13 years ago and apparently worked until recently. Why can't we just accept this partial revert for now and then spend the time necessary to find a better solution? The reason I'm bringing this up is that in its current state 7.0.0/master contains a number of sensible improvements, but this issue makes it impossible to use without a separate patch.